website/pages/docs/k8s/installation/multi-cluster/kubernetes.mdx

```diff
@@ -0,0 +1,388 @@
+   for federation is currently only available in Consul 1.8.0 beta.
+1. `global.tls.enabled` must be `true`. See [Configuring TLS on an Existing Cluster](/docs/k8s/operations/tls-on-existing-cluster)
+   for more information on safely upgrading a cluster to use TLS.
+
+If you've set `enableAutoEncrypt: true`, this is also supported.
+
+1. `global.federation.enabled` must be set to `true`. This is a new config setting.
+1. If using ACLs, you'll already have `global.acls.manageSystemACLs: true`. For the
+   primary cluster, you'll also need to set `global.acls.createReplicationToken: true`.
+   This ensures that an ACL token is created that secondary clusters can use to authenticate
+   with the primary.
+1. Mesh Gateways are enabled with the default configuration. The default configuration
+   creates a Kubernetes Load Balancer service. If you wish to customize the
+   mesh gateway, see the [Helm reference](/docs/k8s/helm#v-meshgateway) for that setting.
+
+With the above settings added to your existing config, follow the [Upgrading](/localhost:3000/docs/k8s/operations/upgrading)
+guide to upgrade your cluster and then come back to the [Federation Secret](#federation-secret) section.
+
+## Federation Secret
+
+The federation secret is a Kubernetes secret containing information needed
+for secondary datacenters/clusters to federate with the primary. This secret is created
+automatically by setting:
+
+```yaml
+global:
+  federation:
+    createFederationSecret: true
+```
+
+After the installation into your primary cluster you will need to export
+this secret:
+
+```sh
+$ kubectl get secret consul-federation -o yaml > consul-federation-secret.yaml
+```
+
+~> **Secret doesn't exist?** If you haven't set `global.name` to `consul` then the name of the secret will
+be your Helm release name suffixed with `-consul-federation` e.g. `helm-release-consul-federation`.
+
+Now you're ready to import the secret into your secondary cluster(s).
+
+Switch `kubectl` context to your secondary Kubernetes cluster. In this example
+our context for our secondary cluster is `dc2`:
+
+```sh
+$ kubectl config use-context dc2
+Switched to context "dc2".
+```
+
+And import the secret:
+
+```sh
+$ kubectl apply -f consul-federation-secret.yaml
+secret/consul-federation configured
+```
+
+#### Federation Secret Contents
+
+The automatically generated federation secret contains:
+
+- **Server certificate authority certificate** - This is the certificate authority
+  used to sign Consul server-to-server communication. This is required by secondary
+  clusters because they must communicate with the Consul servers in the primary cluster.
+- **Server certificate authority key** - This is the signing key for the server certificate
+  authority. This is required by secondary clusters because they need to create
+  server certificates for each Consul server using the same certificate authority
+  as the primary.
```

Comment:
It feels like we should have a security warning box somewhere in this section to make it super clear that access to this secret is equivalent to full-admin privs on the whole Consul install since the TLS cert key effectively allows any attacker to create a server cert that will be accepted by other primary servers who will then replicate all state including ACL tokens and CA keys to the attacker.

Reply:
Similar goes for the gossip token in terms of being able to join the cluster.

Reply:
We typically use the `!> **Security note:**` prefix for these btw.

Reply:
done, thanks I missed that.